### PR TITLE
Remove jquery

### PR DESCRIPTION
**Pull request by ravenclaw900**
Also change carousel library to glider.js.

This tries to keep everything looking the same, replacing most of jquery with native JS and animations with CSS transitions. Also, VSCodium may have changed some formatting on the CSS file, I can revert that if you'd like.

### CODE REVIEW

File: deploy.bash [hunk 1]
```diff
@@ -51,6 +51,8 @@ G_EXEC sed -i "s|<lastmod>.*</lastmod>|<lastmod>$(date '+%Y-%m-%dT%T%:z')</lastm
 
 # 3rd party
 G_EXEC curl -sSf 'https://raw.githubusercontent.com/jquery/codeorigin.jquery.com/main/cdn/jquery-3.6.0.min.js' -o js/jquery.js
```
Comment (MichaIng) [inline deploy.bash:53]:
```suggestion
```

**Comment by MichaIng** [comment]
Many thanks, I always wanted to do this 👍.

> Also, VSCodium may have changed some formatting on the CSS file, I can revert that if you'd like.

No need, not too hard to see the actual changes 😉. In the end everything is minimized anyway. We need to add support to our test deploy API to build from remote repositories as well. Let me implement this for testing.

**Comment by MichaIng** [comment]
Okay, tested it now:
- I don't like the new slider. The button styles can be easily adjusted, of course, also it can be re-packed into a container to align width with other website content. But having the individual elements moving with different delays will be probably more work to put back in. I wonder whether it would be easier and lighter to convert the cslider code from jQuery to plain JS.
- The SBC frames appear instantly, which doesn't feel so well, given that new content appears as well as scroll position changes. A transition time should be added to keep it closer to previous behaviour.
- When selecting a different download filter, the open SBC frame stays open. It should be closed as before, otherwise one cannot see that changed tiles according to the filter.

The change back to native anchor-based scrolling is fine, even that it won't be smooth by default in most browsers. However, due to the navigation bar, the scroll position needs to be higher. Here is how I did it on the survey page: https://github.com/MichaIng/DietPi/blob/c9580d7/.meta/dietpi-survey_report#L1086-L1091

**Comment by ravenclaw900** [comment]
SBC frames should be fixed and transition smoothly now. Scrolling should also be smooth, with `scroll-behavior: smooth` added to the CSS. Scroll position should be fixed now as well.

**Comment by MichaIng** [comment]
Sometimes there seems to be a timing issue when selecting an SBC: Sometimes it doesn't scroll at all, sometimes just a little as of the changed DOM and one time it was scrolling to the top of the page for some reason. I couldn't find the reason, but probably an issue can be when the browser sets the scroll height while the toggleDiv doesn't have it's new height and margin applies yet. In this case it probably helps to trigger the scolling from JS and `return false;` as before, but not sure:

Previously the scrolling was assured to take longer than toggleDiv expansion due to fixed durations. Now the scrolling is handled entirely by the browser, so it might finish before the toggleDiv has finished its expansion, or the other way round, we have no control on it 🤔. I like the approach to use simple browser scrolling and skip all that in JS, and I like the animated toggleDivs, but the way it is here seems to not work well together.

Another thing that is strange is that when hitting the SBCs, it scolls to the top of the toggleDiv instead of to the top of the filter bar, which is the actual `#downloadinfo` element. It seems the `:target::before` does not apply here, but I'm not sure why, since it's common anchor scrolling just like when using the navigation bar, isn't it?

**Comment by MichaIng** [comment]
Marking this as closed in favour of #251. Most parts of `app.js` are very similar, but I want to keep and convert our slider 🙂.

**Pull request closed without merging**